### PR TITLE
Metabolism Precision Runtime Errors

### LINKED ISSUE #5698: New Phoron Salt Chemicals not Working
Injected 30 units of mutone, followed by 30 units of venenum.  Got flavor text about them working, but no mutation and no change in my appearance.  They don't appear to be working.

### PR DESCRIPTION
**Pull request by BurgerLUA**
# Overview
Because of precision issues with floating point values, metabolism would sometimes runtime and cause performance issues. This PR fixes this.

Fixes #5698 as well.

### CODE REVIEW

File: code/modules/reagents/Chemistry-Holder.dm [hunk 1]
```diff
@@ -98,7 +98,6 @@
 //returns 1 if the holder should continue reactiong, 0 otherwise.
 /datum/reagents/proc/process_reactions()
 	if(!my_atom || !my_atom.loc || my_atom.flags & NOREACT)
-		equalize_temperature()
```
Comment (Sindorman) [inline code/modules/reagents/Chemistry-Holder.dm]: Why did you remove this?

Comment (BurgerLUA) [reply code/modules/reagents/Chemistry-Holder.dm]: I thought it was originally part of the problem, but it wasn't, however upon review it should not be there because of ```my_atom.flags & NOREACT```
